### PR TITLE
Support for Link Local Addresses (incl Python Test) and added to CI

### PR DESCRIPTION
**Pull request by yanosz**
Hinweis: Ich kenne die Konfiguration des Travis nicht - damit kann ich auch nicht so recht nachvollziehen, welche Dependencies da ggf. noch nachgelagert werden sollten,.

### CODE REVIEW

File: formatter.py [hunk 1]
```diff
@@ -41,3 +40,14 @@ def finalize(self):
...
         return "\n".join(self.config)
+
+    def is_link_local(self,str):
+        """
+        Checks, if a given string is a valid IPv6 link local Address
+        """
+        try:
+            addr = ipaddress.IPv6Address(str)
+            return addr.is_link_local
+        except:
```
Comment (mweinelt) [inline formatter.py]: Zu breiter Exception Clause.

File: mkbgp [hunk 2]
```diff
@@ -12,7 +12,8 @@ from multiprocessing.dummy import Pool
 
 class BirdFormatter(Formatter):
     "Formatter for bind9 using type forward"
-    def add_data(self, asn, name, template, peer, passive=False):
+    def add_data(self, asn, name, template, peer, interface, passive=False):
+        if self.is_link_local(peer): peer += "%" + interface
```
Comment (mweinelt) [inline mkbgp]: Am besten nach Condition eine newline mit indentation.

File: .travis.yml [hunk 3]
```diff
@@ -2,4 +2,6 @@ language: python
 python:
   - "3.4"
 install: "pip install pep8"
-script: "pep8 --ignore=E501 *.py mk* check netblocks"
+script: 
+  - "pep8 --ignore=E501 *.py mk* check netblocks"
+  - py.test
```
Comment (mweinelt) [inline .travis.yml]: An welcher Stelle installierst du pytest?

**Comment by jplitza** [comment]
The interface is never passed to `add_data()`. Where would it come from anyway? Command line?

And I really don't like that `is_link_local` is a method of `Formatter`. It has nothing to do with formatting, isn't dependent on the formatter or anything. Neither do I understand what kind of exception should be legally caught/ignored there. Simply use `ipaddress.ip_address` to be independent from IPv4/IPv6, and it should work without any exception handling.

**Pull request closed without merging**